### ISSUE TITLE
replace function cannot do double backslashes

### ISSUE THREAD

**Issue by llarsson** [hashicorp/terraform#8709]
I need to use the `replace` function to replace dots in a domain name with two backslashes. Sounds simple, right? Well, it is not. Terraform's replace function does not work entirely as advertised. See the following example:

```
variable "sub_domain" { default = "needs.double.backslashes.com"}

output "with_spaces_works" {
  value = "${replace(var.sub_domain, ".", "\\ \\.")}"
}

output "buggy" {
  value = "${replace(var.sub_domain, ".", "\\\\.")}"
}

output "doubling_does_not_work" {
  value = "${replace(var.sub_domain, ".", "\\\\\\\\.")}"
}
```

I would expect to have the value of "buggy" be `needs\\.double\\.backslashes\\.com`, but it is not. It is in fact `needs\.double\.backslashes\.com`.

The entire output from the example above (save it to a file, run `terraform apply` on it) is:

```
Outputs:

buggy = needs\.double\.backslashes\.com
doubling_does_not_work = needs\.double\.backslashes\.com
with_spaces_works = needs\ \.double\ \.backslashes\ \.com
```

Please fix this. :)

(My particular use case is that I need to feed a string like this into a JSON file and then into a regex parser, so I need to escape the periods twice.)

This is using Terraform 0.7.3.

**Comment by llarsson** [comment]
```
output "buggy_but_with_letters" {
  value = "${replace(var.sub_domain, ".", "AA.")}"
}
```
Shows that had backslash handling been more correct, it would not have been an issue.

Also, removing the period as in the following example is actually apparently a **syntax error**, which makes me believe that backslash handling is rather broken:

```
output "spilling" {
  value = "${replace(var.sub_domain, ".", "\\\\")}"
}
```

**Comment by mitchellh** [comment]
I'll look into this, feels like an HCL bug. Thanks for the easy repros. I haven't tried them yet but they're comprehensive here and give me faith.

**Comment by mitchellh** [comment]
This is fixed and explained here: https://github.com/hashicorp/hcl/pull/150

Unfortunately this has a chance to break working TF configs so we're going to have to hold this until TF 0.8. :( I'm adding it to that milestone so we know. I'm also going to hijack the top of your issue to make a note to my future self.

**Comment by mitchellh** [comment]
We got the fix in, will be part of 0.8 beta 1 this week.
